site/content/en/docs/Advanced/controlling-disruption.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Controlling Disruption"
+date: 2023-01-24T20:15:26Z
+weight: 20
+description: >
+  Game servers running on Agones may be disrupted by Kubernetes; learn how to control disruption of your game servers.
+---
+
+## Disruption in Kubernetes
+
+[A `Pod` in Kubernetes may be disrupted](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#voluntary-and-involuntary-disruptions) for involuntary reasons, e.g. hardware failure, or voluntary reasons, such as when nodes are drained for upgrades. 
+
+By default, Agones assumes your game server should never be disrupted voluntarily and configures the `Pod` appropriately - but this isn't always the ideal setting. Here we discuss how Agones allows you to control the two most significant sources of voluntary `Pod` evictions, node upgrades and Cluster Autoscaler, using the `eviction` API on the `GameServer` object. 
+
+{{< alpha title="`eviction` API" gate="SafeToEvict" >}}
+
+## Benefits of Allowing Voluntary Disruption
+
+It's not always easy to write your game server in a way that allows for disruption, but it can have major benefits:
+
+* Compaction of your cluster using [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) can lead to considerable cost savings for your infrastructure.
+* Allowing automated node upgrades can save you management toil, and lowers the time it takes to patch security vulnerabilites.
+
+## Considerations
+
+When discussing game server pod disruption, it's important to keep two factors in mind:
+
+* **Session Length:** What is your game servers session length (the time from when the `GameServer` is allocated to when its shutdown)? In general, we bucket the session lengths into "less than 10 minutes", "10 minutes to an hour", and "greater than an hour". (See [below](#whats-special-about-ten-minutes-and-one-hour) if you are curious about session length considerations.)
+* **Disruption Tolerance:** Is your game server tolerant of disruption in a timely fashion? If your game server can handle the `TERM` signal and has [terminationGracePeriodSeconds](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution) configured for a period less than 10 minutes, meaning that the game way to control disruption of your game servers that should be applicable to most cloud products with default configurations: [`GameServerSpec.eviction`]({{< ref "/docs/Reference/agones_crd_api_reference.html#agones.dev/v1.GameServerSpec" >}})
```

Comment:
Should this say "meaning that the GameServerSpec.eviction API allows you to control disruption of your game servers on most cloud products with default configurations." or something? The colon followed by the API link is confusing to me.

Reply:
(reworked this section)